File: doc/Programs/LecturePrograms/programs/MCIntro/cpp/program2.cpp
// Particles in a box
#include <iostream>
#include <fstream>
#include <iomanip>
#include "lib.h"
using namespace  std;

ofstream ofile;
int main(int argc, char* argv[])
{
  char *outfilename;
  int initial_n_particles, max_time, time, random_n, nleft; 
  long idum;
  // Read in output file, abort if there are too few command-line arguments
  if( argc <= 1 ){
    cout << "Bad Usage: " << argv[0] <<
      " read also output file on same line" << endl;
    exit(1);
  }
  else{
    outfilename=argv[1];
  }
  ofile.open(outfilename);
  // Read in data 
  cout << "Initial number of particles = " << endl ;
  cin >> initial_n_particles;
  // setup of initial conditions
  nleft = initial_n_particles;
  max_time = 10*initial_n_particles;
  idum = -1;
  // sampling over number of particles
  for( time=0; time <= max_time; time++){
    random_n =  initial_n_particles*ran0(&idum);
    if ( random_n <= nleft){
      nleft -= 1;
    }
    else{
      nleft += 1;
    }
    ofile << setiosflags(ios::showpoint | ios::uppercase);
    ofile << setw(15) << time;
    ofile << setw(15) << nleft << endl;
  }
  return 0; 
} // end main function

File: doc/Programs/LecturePrograms/programs/MCIntro/python/program2.py
# coding=utf-8
#Particles in box example
#Translated to Python by Kyrre Ness Sjøbæk

import sys, numpy.random

#Get name of output file and number of particles in system
if len(sys.argv) == 3:
    filename = sys.argv[1]
    N        = int(sys.argv[2])
else:
    print "Usage:",sys.argv[0],"filename N"
    sys.exit(0)

ofile = open(filename, 'w')

#All particles start in left-hand box
Nl = N

#Do the MC!
for t in xrange(10*N):
    if (int(N*numpy.random.random()) <= Nl):
        Nl -= 1
    else:
        Nl += 1
    ofile.write("%g %g\n" % (t,Nl))

ofile.close()
